Add more docs (#171)
hercules-ci/flake-parts: examples/project-commands/README.md

# project-commands

> **Warning**
> If you copy the flake.nix remember to `git add [-N|--intent-to-add] flake.nix`, otherwise it won't work

This example shows how to create scripts for your project, by leveraging [mission-control](https://github.com/Platonic-Systems/mission-control)

This is a **potential** alternative to:

- Using a `Makefile` to manage your project's scripts

> - Using the popular [Scripts To Rule Them All](https://github.com/github/scripts-to-rule-them-all) approach (having a `scripts/` folder)

Reviewer: ```suggestion
- Using the popular [Scripts To Rule Them All](https://github.com/github/scripts-to-rule-them-all); a naming convention for a `scripts/` directory
```

A bit more info, and folder -> directory, which is the term we generally use in and around Nix.

> - Using a `bin/` folder

Reviewer: ```suggestion
- Using a `bin/` directory
```

## Explanation

In this example we use the [avro-tools](https://avro.apache.org/) to convert our scripts from `.avdl` to `.avsc`.

> You don't need to know anything about avro to understand mission-control and use this example (that's nix baby 🚀).

Reviewer: ```suggestion
You don't need to know anything about avro to understand mission-control and use this example (that's Nix baby 🚀).
```

the commands will be listed as `run build`.

mission-control depends on flake-root, which also exposes the helpful `$FLAKE_ROOT` variable.

After creating the scripts, we need to pass the newly created scripts to the desired shell, in this example we use the default shell.

## Usage

> **Warning**

> > If you copy the flake.nix remember to add it to git, otherwise it won't work

Reviewer 2: Same suggestion here.
